src/Fetcher/fetcher.go

```diff
@@ -0,0 +1,121 @@
+package Fetcher
+
+import (
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func FetchFiles(url, branch, specFile string) error {
+
+	// Check if the .git directory exists in the current directory
+	if _, err := os.Stat("./.git"); err == nil {
+		// Remove the .git directory if it exists (we are still in src which is the only place it'll remove the .git folder)
+		err := os.RemoveAll("./.git")
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create cache dir if it doesn't exist
+	if _, err := os.Stat("cache"); os.IsNotExist(err) {
+		err := os.Mkdir("cache", 0700)
+		errHandler(err)
+	}
+
+	// Pulls files and returns the paths to said files
+    seenFolders := make(map[string]string)
+	paths, err := puller(url, branch, specFile)
+    if err != nil {
+        return err
+    }
+	for _, path := range paths {
+        fileName:= strings.Split(path,"/")
+        if _ , ok := seenFolders[fileName[0]]; !ok {
+            seenFolders[fileName[0]] = ""
+        }
+        os.Rename(path, ("cache/" + fileName[len(fileName)-1]))
+	}
+    for folder, _ := range seenFolders {
+        os.RemoveAll(("./" + folder))
+    }
+    
+	return nil
+}
+
+func errHandler(err error, params ...string) {
+	if err != nil {
+		panic(err.Error() + strings.Join(params, " "))
+	}
+}
+
+func executer(cmd *exec.Cmd) error {
+	//TODO: Figure out a way to take strings as input and build cmd
+	_, err := cmd.CombinedOutput()
+
+	return err
+}
+
+func puller(url, branch, specFile string) ([]string, error) {
+    paths := []string{}
+
+	// Create dummy repo
```

Comment:
Think I'm still confused as to why we have to create a dummy repo? ;-;

Reply:
I'm assuming you mean as to why we create a dummy repo (which is maybe also just bad wording on our end tbf) It just initializes a new git repo with nothing set. This allows us to manipulate the repo using the git CLI commands that we all know and love. Thus we can active sparse checkout and set a whitelist before ever seeing the actual repo we would want  to pull from. This ensures we setup the rules correctly to sparsely and selectively collect the files

Reply:
Ahh okay, that makes sense! We can write your explanation down in the fetcher's docs? I found that it made a lot of sense at least!

Reply:
Feel free to add it to the fetcher docs yes

Reply:
Sure thing! Will be done on a separate branch after the multi-fetcher has been approved :)